backend/de.metas.business/src/main/java/de/metas/invoice/service/impl/AbstractInvoiceDAO.java

```diff
@@ -661,4 +663,35 @@ public Collection<String> retrievePaidInvoiceDocNosForFilter(@NonNull final IQue
 				.map(org.compiere.model.I_C_Invoice::getDocumentNo)
 				.collect(ImmutableList.toImmutableList());
 	}
+
+	@Nullable
+	public I_C_InvoiceLine getOfInOutLine(@Nullable final I_M_InOutLine inOutLine)
+	{
+		if (inOutLine == null)
+		{
+			return null;
+		}
+
+		final ImmutableList<I_C_InvoiceLine> invoiceLineList = queryBL.createQueryBuilder(I_C_InvoiceLine.class)
+				.addEqualsFilter(I_C_InvoiceLine.COLUMNNAME_M_InOutLine_ID, inOutLine.getM_InOutLine_ID())
+				.addEqualsFilter(I_C_InvoiceLine.COLUMNNAME_C_OrderLine_ID, inOutLine.getC_OrderLine_ID())
+				.setJoinOr()
```

Comment:
IMHO this is not respecting the logic from concept

logic as follows:
1st, preserve what it does not, i.e. check SELECT * FROM C_InvoiceLine WHERE M_InOutLine_ID=?
2nd, in case nothing found on 1st, then try M_InOutLIne.C_OrderLine_ID from order line to invoice line

Reply:
i've provided a new version, pls check it/test it